components/brave_wallet_ui/components/extension/sign-panel/cow_swap_order.tsx

```diff
@@ -0,0 +1,249 @@
+
+const makeUnknownToken = (
+  chainId: string,
+  coin: BraveWallet.CoinType,
+  contractAddress: string
+) => ({
+  chainId,
+  coin,
+  symbol: '???',
+  contractAddress,
+  logo: '',
+  isErc721: false,
+  isNft: false,
+  name: '',
+  coingeckoId: UNKNOWN_TOKEN_COINGECKO_ID,
+  decimals: 0
+})
+
+interface Props {
+  data: BraveWallet.SignMessageRequest
+  queueLength: number
+  queueNumber: number
+  isDisabled: boolean
+  onQueueNextSignMessage: () => void
+  onSignIn: () => void
+  onCancel: () => void
+}
+
+export function SignCowSwapOrder(props: Props) {
+  const {
+    data,
+    queueLength,
+    queueNumber,
+    isDisabled,
+    onQueueNextSignMessage,
+    onSignIn,
+    onCancel
+  } = props
+
+  // State
+  const [showDetails, setShowDetails] = React.useState<boolean>(false)
+
+  const { data: combinedTokensList } = useGetCombinedTokensListQuery()
+  const { data: accounts } = useGetAccountInfosRegistryQuery()
+
+  const cowSwapOrder = data.signData.ethSignTypedData?.meta?.cowSwapOrder
+
+  const buyToken = cowSwapOrder?.buyToken
+    ? findTokenByContractAddress(cowSwapOrder.buyToken, combinedTokensList)
+    : undefined
+  const sellToken = cowSwapOrder?.sellToken
+    ? findTokenByContractAddress(cowSwapOrder.sellToken, combinedTokensList)
+    : undefined
+
+  const { data: network } = useGetNetworkQuery({
+    chainId: data.chainId,
+    coin: data.coin
+  })
+
+  const senderLabel = accounts && getAccountLabel(data.accountId, accounts)
+  const recipientLabel =
+    accounts &&
+    cowSwapOrder &&
+    cowSwapOrder.receiver &&
+    getAddressLabel(cowSwapOrder.receiver, accounts)
+  const senderOrb = useAccountOrb({
+    accountId: data.accountId,
+    address: data.accountId.address
+  })
+  const recipientOrb = useAddressOrb(cowSwapOrder?.receiver, { scale: 10 })
+
+  return (
+    <StyledWrapper>
+      <TopRow>
+        <NetworkText />
+        <TransactionQueueSteps
+          queueNextTransaction={onQueueNextSignMessage}
+          transactionQueueNumber={queueNumber}
+          transactionsQueueLength={queueLength}
+        />
+      </TopRow>
+
+      <HeaderTitle>{getLocale('braveWalletSwapReviewHeader')}</HeaderTitle>
+
+      <Origin originInfo={data.originInfo} />
+
+      {!showDetails && (
+        <SwapBase
+          sellToken={
+            sellToken || (cowSwapOrder
+              ? makeUnknownToken(
+                  data.chainId,
+                  data.coin,
+                  cowSwapOrder.sellToken
+                )
+              : undefined)
+          }
+          buyToken={
+            buyToken || (cowSwapOrder
+              ? makeUnknownToken(data.chainId, data.coin, cowSwapOrder.buyToken)
+              : undefined)
+          }
+          sellAmount={cowSwapOrder?.sellAmount}
+          buyAmount={cowSwapOrder?.buyAmount}
+          senderLabel={senderLabel}
+          senderOrb={senderOrb}
+          recipientOrb={recipientOrb}
+          recipientLabel={recipientLabel}
```

Comment:
Probably out of scope for this PR, but, one thing that would help with the security is to show a warning if the `receiver` account and the `signing` account is not the same (or is not 0x0, since that in the smart contract refers to the owner of the funds).

Changing the receiver is a great CoW Swap feature, it allows you to SWAP and send in one operation. However, i think is nice to make the user aware the destination of the funds is different than the origin of them

Reply:
Indeed that's a good point. The addresses should also link to the explorer. Will take inputs from the design team for a future iteration.

Reply:
++ seems like a good follow up item